packages/flow-core/src/components/f-tag/f-tag.ts

```diff
@@ -0,0 +1,235 @@
+import { html, unsafeCSS } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import { FRoot } from "../../mixins/components/f-root/f-root";
+import eleStyle from "./f-tag.scss";
+import { unsafeSVG } from "lit-html/directives/unsafe-svg.js";
+import loader from "../../mixins/svg/loader";
+import { classMap } from "lit-html/directives/class-map.js";
+import validateColor from "validate-color";
+
+/**
+ * @summary Buttons allow users to perform an action or to initiate a new function.
+ */
+@customElement("f-tag")
+export class FTag extends FRoot {
+  /**
+   * css loaded from scss file
+   */
+  static styles = [unsafeCSS(eleStyle)];
+
+  /**
+   * @attribute label property defines the text label on a tag. Label of a tag is always uppercase.
+   */
+  @property({ type: String })
+  label!: string;
+
+  /**
+   * @attribute category of tag
+   */
+  @property({ reflect: true, type: String })
+  fill?: string;
+
+  /**
+   * @attribute The medium size is the default and recommended option.
+   */
+  @property({ reflect: true, type: String })
+  size?: "large" | "medium" | "small" = "medium";
+
+  /**
+   * @attribute The states on tags are to indicate various degrees of emphasis of the action.
+   */
+  @property({ reflect: true, type: String })
+  state?: "primary" | "neutral" | "success" | "warning" | "danger" | "inherit";
+
+  /**
+   * @attribute Icon-left enables an icon on the left of the label of a tag.
+   */
+  @property({ reflect: true, type: String, attribute: "icon-left" })
+  iconLeft?: string;
+
+  /**
+   * @attribute Icon-right enables an icon on the right of the label of a tag.
+   */
+  @property({ reflect: true, type: String, attribute: "icon-right" })
+  iconRight?: string;
+
+  /**
+   * @attribute Counter property enables a counter on the tag.
+   */
+  @property({ reflect: true, type: Number })
+  counter?: string;
+
+  /**
+   * @attribute Loader icon replaces the content of the tag .
+   */
+  @property({ reflect: true, type: Boolean })
+  loading?: boolean = false;
+
+  /**
+   * @attribute Loader icon replaces the content of the tag .
+   */
+  @property({ reflect: true, type: Boolean })
+  selected?: boolean = false;
+
+  /**
+   * @attribute The disabled attribute can be set to keep a user from clicking on the tag.
+   */
+  @property({ reflect: true, type: Boolean })
+  disabled?: boolean = false;
+
+  /**
+   * compute icon size based on tag size
+   */
+  get iconSize() {
+    if (this.size === "large") {
+      return "medium";
+    } else if (this.size === "medium") {
+      return "small";
+    } else {
+      return "x-small";
+    }
+  }
+
+  /**
+   * render on playground
+   */
+  connectedCallback() {
```

Comment:
Why`resize` required here?
is tag has responsive behaviour which executes some javascript?